src/main/scala/com/comcast/xfinity/sirius/uberstore/segmented/SegmentedCompactor.scala

```diff
@@ -92,17 +105,19 @@ object SegmentedCompactor {
   }
 
   /**
-   * Compact a single Segment, according to the GC keys provided
+   * Compact a single Segment, according to the GC keys provided.  Deletes that are older than the maxDeleteAge
+   * are purged.
    *
    * @param keys keys to gc against
    * @param source source segment, to be GC'ed
    * @param dest destination segment, empty, will be populated
    */
   private def compactSegment(keys: Set[String], source: Segment, dest: Segment) {
-    // XXX if we want to remove old deletes from the log...
-    // val currentTime = System.currentTimeMillis
+    val currentTime = System.currentTimeMillis
+    val earliestDelete = if (maxDeleteAgeMillis > currentTime) 0L else currentTime - maxDeleteAgeMillis
```

Comment:
This could use a comment. I guess none of the updates are likely to come from the 60's, so we're probably not missing any deletes with this, but it's a little confusing, and I'm not sure it makes this any safer/simpler/easier. We could probably remove the `if` statement and just keep it simply `currentTime - maxDeleteAgeMillis`